docs/static/docker.asciidoc

```diff
@@ -0,0 +1,114 @@
+configuration via the +docker run+ command:
+
+["source","sh",subs="attributes"]
+--------------------------------------------
+docker run --rm -it -v ~/pipeline/:/usr/share/logstash/pipeline/ {docker-image}
+--------------------------------------------
+
+Every file in the host directory +~/pipeline/+ will then be parsed
+by Logstash as pipeline configuration.
+
+If you don't provide configuration to Logstash, it will run with a
+minimal config that listens for messages from the
+<<plugins-inputs-beats,Beats input plugin>> and echoes any that are
+received to `stdout`. In this case, the startup logs will be similar
+to the following:
+
+["source","text"]
+--------------------------------------------
+Sending Logstash logs to /usr/share/logstash/logs which is now configured via log4j2.properties.
+[2016-10-26T05:11:34,992][INFO ][logstash.inputs.beats    ] Beats inputs: Starting input listener {:address=>"0.0.0.0:5044"}
+[2016-10-26T05:11:35,068][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>500}
+[2016-10-26T05:11:35,078][INFO ][org.logstash.beats.Server] Starting server on port: 5044
+[2016-10-26T05:11:35,078][INFO ][logstash.pipeline        ] Pipeline main started
+[2016-10-26T05:11:35,105][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
+--------------------------------------------
+
+This is the default configuration for the image, defined in
++/usr/share/logstash/pipeline/logstash.conf+.  If this is the
+behaviour that you are observing, ensure that your pipeline
+configuration is being picked up correctly, and that you are replacing
+either +logstash.conf+ or the entire +pipeline+ directory.
+
+==== Settings Files
+
+Settings files can also be provided through bind-mounts. Logstash
+expects to find them at +/usr/share/logstash/config/+.
+
+It's possible to provide an entire directory containing all needed
+files:
+
+["source","sh",subs="attributes"]
+--------------------------------------------
+docker run --rm -it -v ~/settings/:/usr/share/logstash/config/ {docker-image}
+--------------------------------------------
+
+Alternatively, a single file can be mounted:
+
+["source","sh",subs="attributes"]
+--------------------------------------------
+docker run --rm -it -v ~/settings/logstash.yml:/usr/share/logstash/config/logstash.yml {docker-image}
+--------------------------------------------
+
+==== Custom Images
+
+Bind-mounted configuration is not the only option, naturally. If you
+prefer the _Immutable Infrastructure_ approach, you can prepare a
+custom image containing your configuration by using a +Dockerfile+
+like this one:
+
+["source","dockerfile",subs="attributes"]
+--------------------------------------------
+FROM {docker-image}
+RUN rm -f /usr/share/logstash/pipeline/logstash.conf
+ADD pipeline/ /usr/share/logstash/pipeline/
```

Comment:
I don't know. It doubles the size of the example by filling it with things that aren't directly relevant to the point.

```
FROM docker.elastic.co/logstash/logstash:5.0.1
RUN rm -f /usr/share/logstash/pipeline/logstash.conf
ADD pipeline/ /usr/share/logstash/pipeline/
ADD config/ /usr/share/logstash/config/
USER root
RUN chown -R logstash:logstash /usr/share/logstash/pipeline/ /usr/share/logstash/config/
RUN chmod 0755 /usr/share/logstash/pipeline/ /usr/share/logstash/config/
RUN chmod 0444 /usr/share/logstash/pipeline/* /usr/share/logstash/config/*
USER logstash
```

What a mess! It obscures the topic of "how to configure Logstash on Docker" with a discussion of "how to manage file ownership and modes in Docker". 

It forces you to switch user (twice) and become very explicit about file permissions, because you now have to be very sure that the `logstash` user _doesn't_ have write access to the files. Logstash is, among other things, capable of [executing arbitrary code](https://www.elastic.co/guide/en/logstash/current/plugins-filters-ruby.html#plugins-filters-ruby-code), so giving the process write access to its own files seems like something you want to avoid. A simple `chown` therefore, is insufficient. If we teach `chown`, we have to teach `chmod` and `USER` too. Maybe all this belongs in a dedicated "Securing Logstash" guide, but I don't think it belongs in the Docker documentation.